src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GetActionPlanSummariesTest.kt

```diff
@@ -0,0 +1,155 @@
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.aValidTokenWithNoAuthorities
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.aValidTokenWithViewAuthority
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.anotherValidPrisonNumber
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.bearerToken
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.ActionPlanSummaryListResponse
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.CreateActionPlanRequest
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.ErrorResponse
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.aValidActionPlanSummaryResponse
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.aValidCreateActionPlanRequest
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.aValidGetActionPlanSummariesRequest
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.assertThat
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.withBody
+import java.time.LocalDate
+
+class GetActionPlanSummariesTest : IntegrationTestBase() {
+
+  companion object {
+    private const val URI_TEMPLATE = "/action-plans"
+  }
+
+  @Test
+  fun `should return unauthorized given no bearer token`() {
+    webTestClient.post()
+      .uri(URI_TEMPLATE)
+      .exchange()
+      .expectStatus()
+      .isUnauthorized
+  }
+
+  @Test
+  fun `should return forbidden given bearer token without required role`() {
+    // Given
+    val request = aValidGetActionPlanSummariesRequest()
+
+    // When
+    val response = webTestClient.post()
+      .uri(URI_TEMPLATE)
+      .withBody(request)
+      .bearerToken(aValidTokenWithNoAuthorities(privateKey = keyPair.private))
+      .exchange()
+      .expectStatus()
+      .isForbidden
+      .returnResult(ErrorResponse::class.java)
+
+    // Then
+    val actual = response.responseBody.blockFirst()
+    assertThat(actual)
+      .hasStatus(FORBIDDEN.value())
+      .hasUserMessage("Access Denied")
+      .hasDeveloperMessage("Access denied on uri=/action-plans")
+  }
+
+  @Test
+  fun `should get empty list of action plan summaries given prisoner has no action plan`() {
+    // Given
+    val request = aValidGetActionPlanSummariesRequest(prisonNumbers = listOf(aValidPrisonNumber()))
+
+    // When
+    val response = webTestClient.post()
+      .uri(URI_TEMPLATE)
+      .withBody(request)
+      .bearerToken(aValidTokenWithViewAuthority(privateKey = keyPair.private))
+      .contentType(MediaType.APPLICATION_JSON)
+      .exchange()
+      .expectStatus()
+      .isOk
+      .returnResult(ActionPlanSummaryListResponse::class.java)
+
+    // Then
+    val actual = response.responseBody.blockFirst()
+    assertThat(actual).hasEmptySummaries()
+  }
+
+  @Test
```

Comment:
Minor, and perhaps its not an issue; but we know that integration tests are slow - do we need 3 happy path tests?
I think we need the one that returns many summaries .... _maybe_ we need the empty collection one? (not sure); but I'm not sure we need the one that returns 1 summary .... What do you think? Does the service tests cover returning 0, 1 and many ?

Reply:
Happy to remove the single one 👍